src/cpp/session/modules/SessionCodeSearch.cpp

```diff
@@ -69,18 +69,27 @@ namespace code_search {
 
 namespace {
 
-bool isInCmakeBuildDirectory(const FilePath& filePath)
+bool isWithinIgnoredDirectory(const FilePath& filePath)
 {
-   FilePath parentPath = filePath.parent();
-   while (!parentPath.empty()
-          && parentPath != projects::projectContext().directory())
+   for (FilePath parentPath = filePath.parent();
+        !parentPath.empty();
+        parentPath = parentPath.parent())
```

Comment:
1. Did you test this on Windows? I can't recall whether it has a root directory which has a parent that resolves to the empty string (obviously if it resolves to anything else that'd be bad here)

2. Any reason this does not terminate at the project directory as the original code did?